src/main/kotlin/com/cultureamp/kafka/connect/plugins/transforms/UnifyLegacySlackIntegrationPayload.kt

```diff
@@ -0,0 +1,104 @@
+package com.cultureamp.kafka.connect.plugins.transforms
+
+import org.apache.kafka.common.config.ConfigDef
+import org.apache.kafka.connect.connector.ConnectRecord
+import org.apache.kafka.connect.data.Field
+import org.apache.kafka.connect.data.Schema
+import org.apache.kafka.connect.data.SchemaBuilder
+import org.apache.kafka.connect.data.Struct
+import org.apache.kafka.connect.errors.DataException
+import org.apache.kafka.connect.transforms.Transformation
+import org.apache.kafka.connect.transforms.util.Requirements
+
+class UnifyLegacySlackIntegrationPayload<R : ConnectRecord<R>> : Transformation<R> {
+    val attributesToIgnoreList = arrayListOf("account_aggregate_id", "oauth_response_data.access_token", "oauth_response_data.team_id", "oauth_response_data.team_name", "oauth_response_data.bot.bot_access_token", "oauth_response_data.team.id", "oauth_response_data.team.name")
+    val PURPOSE = "unify legacy slack integration data"
+    override fun configure(configs: MutableMap<String, *>?) {}
+
+    override fun close() {}
+
+    fun removeIgnoredAttributes(fields: List<Field>, builder: SchemaBuilder, hierarchy: String = ""): SchemaBuilder {
+        for (field in fields) {
+            if ("$hierarchy${field.name()}" !in attributesToIgnoreList) {
+                if (field.schema().type().getName() == "struct") {
+                    val childSchema = removeIgnoredAttributes(field.schema().fields(), SchemaBuilder.struct(), "$hierarchy${field.name()}.")
+                    // Only add child schema if is not empty
+                    if (childSchema.fields().count() > 0) {
+                        builder.field(field.name(), childSchema.build())
+                    }
+                } else {
+                    builder
+                        .field(field.name(), field.schema())
+                }
+            }
+        }
+        return builder
+    }
+
+    fun populateValue(originalValues: Struct, updatedValues: Struct): Struct {
+        var newFields = updatedValues.schema().fields()
+        for (field in newFields) {
+            try {
+                if (field.schema().type().getName() == "struct") {
+                    val childValue = populateValue(Requirements.requireStruct(originalValues.get(field.name()), PURPOSE), Struct(field.schema()))
+                    updatedValues.put(field.name(), childValue)
+                } else {
+                    updatedValues
+                        .put(field.name(), originalValues.get(field.name()))
+                }
+            } catch (e: DataException) {
```

Comment:
Since nothing is done in `catch` block, the whole `try/catch` statements can be removed

Reply:
Ahh... we need the catch as `.get(field.name())) ` will throw an exception **IF** `field.name()` does not exists

Do we need to log? Probably not as I do except some field does not exists as we are manipulating the schema